src/main/java/be/syntra/devshop/DevshopFront/controllers/SearchController.java

```diff
@@ -9,48 +9,128 @@
-        // todo: (DEV-015) the searchTerm is retrieved from the from the SearchModelDto here
-        List<Product> productList = productListCacheService.findBySearchRequest(searchService.getSearchModel().getSearchRequest()).getProducts();
-        model.addAttribute("products", productList);
-        model.addAttribute("cart", cartService.getCart());
-        return "product/productOverview";
+        searchService.setSearchRequest(searchRequest);
+        List<Product> productList = applySearch(true);
+        return setTemplateModel(model, productList);
+    }
+
+    @GetMapping("/pricelow/")
+    public String searchPriceLow(@RequestParam String priceLow, Model model) {
+        searchService.setPriceLow(new BigDecimal(priceLow));
+        List<Product> productList = applySearch(true);
+        return setTemplateModel(model, productList);
+    }
+
+    @GetMapping("/pricehigh/")
+    public String searchPriceHigh(@RequestParam String priceHigh, Model model) {
+        searchService.setPriceHigh(new BigDecimal(priceHigh));
+        List<Product> productList = applySearch(true);
+        return setTemplateModel(model, productList);
+    }
+
+    @GetMapping("/description/")
+    public String searchProductDescription(@RequestParam String description, Model model) {
+        searchService.setDescription(description);
+        List<Product> productList = applySearch(true);
+        return setTemplateModel(model, productList);
+    }
+
+    @GetMapping("/sortbyname")
+    public String sortByName(Model model) {
+        List<Product> productList = applySearch(searchService.getSearchModel().isSearchResultView());
+        return getProductsSortedByName(model, productList);
+    }
+
+    @GetMapping("/sortbyprice")
+    public String sortByPrice(Model model) {
+        List<Product> productList = applySearch(searchService.getSearchModel().isSearchResultView());
+        return getProductsSortedByPrice(model, productList);
+    }
+
+    @GetMapping("/archived/sortbyname")
+    public String sortArchivedByName(Model model) {
+        List<Product> productList = productService.findAllArchived().getProducts();
+        return getProductsSortedByName(model, productList);
+    }
+
+    @GetMapping("/archived/sortbyprice")
+    public String sortArchivedByPrice(Model model) {
+        List<Product> productList = productService.findAllArchived().getProducts();
+        return getProductsSortedByPrice(model, productList);
+    }
+
+    private List<Product> applySearch(boolean isSearchResultView) {
+        searchService.setSearchResultView(isSearchResultView);
+        searchService.setArchivedView(false);
+        List<Product> productList = productListCacheService.findBySearchRequest(searchService.getSearchModel()).getProducts();
+        List<Product> filteredList = productListCacheService.filterByPrice(productList, searchService.getSearchModel()).getProducts();
+        return (null != searchService.getSearchModel().getDescription())
```

Comment:
boolean hasProductDescription() is a more readable if statement.

voorbeeld: bindingResult.hasErrors() -> same solution for a null-check.

Reply:
extracted